quicktests/overlaying/tests/functional/axisLabelAt.js

```diff
@@ -0,0 +1,50 @@
+function makeData() {
+  "use strict";
+
+  var  data1 = [{month: "January", avg: 2.75, city: "Palo Alto"}, {month: "February", avg: 3.07, city: "Palo Alto"}, {month: "March", avg: 2.26, city: "Palo Alto"}, {month: "April", avg: 0.98, city: "Palo Alto"}];
+  var  data2 = [{month: "January", avg: 4.21, city: "San Francisco"}, {month: "February", avg: 4.10, city: "San Francisco"}, {month: "March", avg: 2.74, city: "San Francisco"}, {month: "April", avg: 1.18, city: "San Francisco"}];
+  var  data3 = [{month: "January", avg: 2.99, city: "San Jose"}, {month: "February", avg: 3.32, city: "San Jose"}, {month: "March", avg: 2.04, city: "San Jose"}, {month: "April", avg: 1.06, city: "San Jose"}];
+
+  return [data1, data2, data3];
+}
+
+function run(svg, data, Plottable){
+  "use strict";
+
+  var xScale = new Plottable.Scales.Category();
+  var yScale = new Plottable.Scales.Linear();
+  var colorScale = new Plottable.Scales.Color();
+
+  var xAxis = new Plottable.Axes.Category(xScale, "bottom");
+  var yAxis = new Plottable.Axes.Numeric(yScale, "left");
+
+  var clusteredPlot = new Plottable.Plots.ClusteredBar("vertical")
+    .addDataset(new Plottable.Dataset(data[0]))
+    .addDataset(new Plottable.Dataset(data[1]))
+    .addDataset(new Plottable.Dataset(data[2]))
+    .x(function(d) { return d.month; }, xScale)
+    .y(function(d) { return d.avg; }, yScale)
+    .attr("label", function(d) { return d.avg; })
+    .attr("fill", function(d) { return d.city; }, colorScale);
+
+  var title = new Plottable.Components.TitleLabel("Click on a axis label");
+
+  var chart = new Plottable.Components.Table([
+                                            [null,  title],
+                                            [yAxis, clusteredPlot],
+                                            [null,  xAxis]
+                                            ]);
+
+  [xAxis, yAxis].forEach((axis) => {
+    var clickInteraction = new Plottable.Interactions.Click();
+    clickInteraction.onClick((point) => {
+        const label = JSON.stringify(axis.tickLabelAt(point));
```

Comment:
This sequence of events feels very roundabout:

1. user clicks
2. our click listener (from mouseDispatcher) creates a "point" object from the `event.clientX` and `event.clientY`
3. axis.tickLabelAt iterates through all the tick label DOM elements and linear scans to find which one was clicked on
4. We get the data corresponding to that label

Why can't we do:

1. user clicks
2. We get the tick label DOM element we clicked on from `event.target`
3. We get the data corresponding to that label

Actually maybe we can add both? `tickLabelAt` and `tickLabelInteractedWith`?